pkg/pillar/cmd/tpmmgr/tpmmgr.go

```diff
@@ -908,6 +936,109 @@ func createEcdhCert() error {
 	return nil
 }
 
+// create Ecdh Template
+func createEcdhSoftTemplate() x509.Certificate {
+	serialNumberLimit := new(big.Int).Lsh(big.NewInt(1), 128)
+	serialNumber, _ := rand.Int(rand.Reader, serialNumberLimit)
+	notBefore := time.Now()
+	// twenty years
+	notAfter := notBefore.AddDate(20, 0, 0)
+
+	template := x509.Certificate{
+		SerialNumber: serialNumber,
+		Subject: pkix.Name{
+			Country:      []string{"US"},
+			Province:     []string{"CA"},
+			Locality:     []string{"Santa Clara"},
+			Organization: []string{"Zededa, Inc"},
+			CommonName:   "Cipher Block",
+		},
+		NotBefore: notBefore,
+		NotAfter:  notAfter,
+
+		KeyUsage:              x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature,
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
+		BasicConstraintsValid: true,
+	}
+	return template
+}
+
+// get the file based software ECDH private key
```

Comment:
From the comment, name, and code I can't tell what his function does.
Code and comment seems to imply a get, and if the get succeeds it generates a key.

I think part of the naming issue is that getEveNodePrivateKey is really getting the debice private key.
We call that a deviceKey and not an EveNode key. 
Plus this comment should say "generate the software ECDH key and certificate, where the certificate is signed using the device privaye key".